hathor/verification/merge_mined_block_verifier.py

```diff
@@ -12,8 +12,14 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
+from hathor.transaction import MergeMinedBlock
 from hathor.verification.block_verifier import BlockVerifier
 
 
 class MergeMinedBlockVerifier(BlockVerifier):
     __slots__ = ()
+
+    def verify_aux_pow(self, block: MergeMinedBlock) -> None:
```

Comment:
No... Because calling `MergeMinedBlockVerifier.verify_without_storage()` will inherit the call from `BlockVerifier.verify_without_storage()`, which simply calls `block.verify_without_storage()`. When that `block` instance is a `MergeMinedBlock`, its own `verify_without_storage()` will be called, which includes the call to `verify_aux_pow()`.

However, I do agree that this is a bit confusing. This is just because this is an intermediary step. In [part 3](https://github.com/HathorNetwork/hathor-core/pull/798/files), `verify_without_storage()` is moved to `MergeMinedBlockVerifier`. After that, I want to remove all inheritance from verifiers, which will improve clarity and prevent errors, in my opinion (that's still not implemented).

Reply:
Got it. I agree it's confusing. I feel we should add a task to double check that all verifications are being correctly called at the end of this sequence of refactors.